principles/solid/dependencyInjection.md

```diff
@@ -1,22 +1,68 @@
 ## Dependency Injection
 
-[TODO]
+The principle of dependency inversion refers to the decoupling of software modules. 
+This way, instead of high-level modules depending on low-level modules, both will depend on abstractions.
+
+Consider the example below. We have a Car class that depends on the concrete Engine class; therefore, it is not obeying DIP.
 
 ### Bad example
 
-[TODO]
+    public class Car {
+        private Engine engine;
+
+        public Car(Engine e) {
+            engine = e;
+        }
+
+        public void start() {
+            engine.start();
+        }
+    }
+
+    public class Engine {
+        public void start() {...}
+    }
+
+The code will work, for now, but what if we wanted to add another engine type, let’s say a diesel engine? 
```

Comment:
This is not the right example for Dependency Injection. It's more familiar with Open/Closed